app/views/bgeigie_imports/_head.html.erb

```diff
@@ -5,33 +5,69 @@
       <div id="current-status"><%= content_for(:current_status) %></div>
     <%- end -%>
     <ul class="inline nav-stacked pull-right">
-      <%- if !@bgeigie_import.rejected? && !@bgeigie_import.approved? && @bgeigie_import.ready_for_submission? && user_signed_in? && @bgeigie_import.processed? && (current_user.moderator? || @bgeigie_import.user_id == current_user.id ) -%>
-        <li>
-          <%= form_for [:reject, @bgeigie_import] do |f| %>
-            <%= f.submit t('.reject'), :class => 'btn btn-primary' %>
+      <%- if user_signed_in? -%>
+        <%# buttons in this block are logged-in users %>
+        <%- if current_user.moderator? -%>
+          <%# buttons in this block such as "Approve", "Reject" and "Fix Drive" are for only moderator %>
+          <%- unless @bgeigie_import.rejected? -%>
+            <%# buttons are shown only if import has not been rejected %>
+            <%- if @bgeigie_import.approved? -%>
+              <%# if import has been approved, show "Fix Drive" button %>
+              <li>
+                <%= form_for [:fixdrive, @bgeigie_import] do |f| %>
+                  <%= f.submit t('.fixdrive'), class: 'btn btn-primary' %>
+                <%- end -%>
+              </li>
+            <%- else %>
+              <%- case @bgeigie_import.status -%>
+              <%- when 'processed' -%>
+                <%# when import has been processed, moderator could reject %>
+                <li>
+                  <%= form_for @bgeigie_import, url: { action: 'reject' } do |f| %>
```

Comment:
Nice improvements here. Got me wondering if we should have a button helper as well rather than all these little `form_for` blocks.

Reply:
indeed. created helper